pkg/services/hcloud/server/server_test.go

```diff
@@ -339,3 +340,90 @@ var _ = Describe("Test ValidateLabels", func() {
 		}),
 	)
 })
+
+var _ = Describe("Test handleRateLimit", func() {
```

Comment:
the most important part to test is whether a condition is set on the object or not. You don't test that so far